src/std/misc/string.ss

```diff
@@ -215,3 +218,29 @@
 	(string-set! str i (random-word-char)))
       str)
     ""))
+
+;; str converts all of its arguments into a single string.
+;; When called without an argument an empty string is returned.
+;;
+;; Examples:
+;;  (str 2.0)               => "2.0"
+;;  (str "hello" ", world") => "hello, world"
+(def* str
+  ((v) (if (string? v) v
+           (format ((make-str-fmt) v) v)))
+  (xs (call-with-output-string
+       (lambda (port)
+         (let (fmt-proc (make-str-fmt))
+           (let loop ((rest xs))
+             (when (pair? rest)
+               (let (v (car rest))
+                 (if (string? v)
+                   (write-string v port)
+                   (fprintf port (fmt-proc v) v)))
+               (loop (cdr rest)))))))))
```

Comment:
I would write this using `match`, it is more idiomatic (and faster in safe context because the destructuring can utilize unsafe ops after the predicate check.
So:
```
(match rest
 ([v . rest]
  (if (string? v)
    (write-string v port)
    (fprintf port (str-format v) v)))
  (loop rest))
 (else (void)))
```

Reply:
I prefer `match` too, but thought that it would be "slower" than the `if`s.

Reply:
It's not, it will actually be faster because it ellides duplicates checks in the destructuring.